CMakeLists.txt

```diff
@@ -77,6 +77,13 @@ include(FindPkgConfig)
 # Keep consistent with Cargo files that pull in glib: `git grep glib -- '*/Cargo.toml'`
 pkg_check_modules(GLIB REQUIRED IMPORTED_TARGET glib-2.0>=2.58)
 
+## show a warning if Go is installed and is version 1.21.x
+execute_process(COMMAND bash -c "go version | { read _ _ v _; echo \${v#go}; }" OUTPUT_VARIABLE GO_VERSION OUTPUT_STRIP_TRAILING_WHITESPACE ERROR_QUIET)
+if(GO_VERSION MATCHES "^1.21.")
+    # https://github.com/shadow/shadow/issues/3267
+    MESSAGE(WARNING "Go version '${GO_VERSION}' is not supported. Some extra tests will likely fail. See issue #3267.")
```

Comment:
Since this is being checked even when `--extra` is not specified, perhaps we want to change the warning to indicate that tests will only fail if trying to run the extra tests?

Something like:

```diff
- Some extra tests will likely fail.
+ If running the extra tests, the go-based extra tests will likely fail.
```

Reply:
Sounds good, changed it to "If running Shadow's extra tests, the Go-based extra tests will fail."